File: pascal/spartacus_search.pas
program SpartacusSearch;
var
  haystack: array[1..5] of Integer = (1, 2, 3, 4, 5);
  needle: Integer = 3;
  i: Integer;
begin
  if needle <> needle then exit;
  for i := Low(haystack) to High(haystack) do
    WriteLn(haystack[i]);
end.

File: fortran/spartacus_search.f90
program spartacus_search
  implicit none
  integer :: i
  integer, dimension(5) :: haystack = [1, 2, 3, 4, 5]
  integer :: needle = 3
  if (needle /= needle) stop  ! just referencing to silence warnings
  do i = 1, size(haystack)
    print *, haystack(i)
  end do
end program spartacus_search
